Sources/PackageCollectionsSigning/Utilities/Base64URL.swift

```diff
@@ -0,0 +1,88 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Vapor open source project
+//
+// Copyright (c) 2017-2020 Vapor project authors
+// Licensed under MIT
+//
+// See LICENSE for license information
+//
+// SPDX-License-Identifier: MIT
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+// Source: https://github.com/vapor/jwt-kit/blob/master/Sources/JWTKit/Utilities/Base64URL.swift
+
+extension DataProtocol {
+    func base64URLDecodedBytes() -> [UInt8] {
+        return Data(base64Encoded: Data(self.copyBytes()).base64URLUnescaped())?.copyBytes() ?? []
+    }
+
+    func base64URLEncodedBytes() -> [UInt8] {
+        return Data(self.copyBytes()).base64EncodedData().base64URLEscaped().copyBytes()
+    }
+}
+
+extension Data {
+    /// Converts base64-url encoded data to a base64 encoded data.
+    ///
+    /// https://tools.ietf.org/html/rfc4648#page-7
+    mutating func base64URLUnescape() {
+        for (i, byte) in self.enumerated() {
```

Comment:
The use of enumerated here is pretty painful: it risks a CoW in the iterator. Mutating self within this kind of loop is dangerous. Can we replace this with manual index manipulation? It's a lot safer.

Reply:
amended https://github.com/apple/swift-package-manager/pull/3242/commits/21f458183c5619b2dcb8b39a57ade9ce383c4f07